MODULE.bazel

```diff
@@ -0,0 +1,181 @@
+module(
+    name = "io_bazel_rules_closure",
+    version = "0.15.0",
+)
+
+bazel_dep(
+    name = "bazel_skylib",
+    version = "1.7.1",
+)
+
+bazel_dep(
+    name = "platforms",
+    version = "0.0.5",
+)
+
+# TODO(mollyibot): Remove this in the future.
+bazel_dep(
+    name = "rules_proto",
+    version = "7.0.2",
+)
+
+bazel_dep(
+    name = "protobuf",
+    version = "29.1",
+    repo_name = "com_google_protobuf",
+)
+
+#TODO(mollyibot):Remove this after switching to new protobuf release
+bazel_dep(
+    name = "abseil-cpp",
+    version = "20240722.0",
+)
+
+bazel_dep(name = "google_bazel_common")
+
+git_override(
+    module_name = "google_bazel_common",
```

Comment:
Isn't `google_bazel_common` just re-exporting some maven java targets? Should it just be deprecated and other projects just depend on Maven targets fetched by rules_jvm_external directly?

Reply:
It generally does couple of things; 
 - aligns internal google3 targets with external targets to make it easy to maintain mapping
 - handles more complicated setups like Dagger and AutoValue where Java annotation processing tools are involved.
 - update targets generally in a compatible way so open-source Google projects doesn't need to think about that aspect.
 
cc. @cpovirk

Reply:
Ah I see, that makes sense. 

/cc @Wyverald This is probably related to our effort to make open-sourcing google projects easier.

Reply:
(Yes, I think I saw some a recommendation from the Copybara team recently to follow a model that "aligns internal google3 targets with external targets to make it easy to maintain mapping." Naturally, I don't remember where :) I don't personally have a great feel for what external mirroring of `BUILD` files looks like _without_ bazel-common, but _with_ bazel-common, it often just works, though there are certainly exceptions....)

Reply:
From a brief look, `google_bazel_common` kind of looks like Bazel's own `third_party` directory, where we provide some BUILD files for some Java libraries we use. So I can definitely see the value there.

Conversely, it has a few problems -- mainly that it's a big clump of dependencies (using one requires using all others) and that the name is very unclear (does not indicate that it only concerns Java projects, and also there isn't really anything Google-specific in there (CMIIW)).

I think we should try to submit some modules to BCR that somehow bring in the jars from Maven while also providing a BUILD file patch, just like we do for many C++ libraries. IMO that would obviate usage of `google_bazel_common`.

Reply:
> I think we should try to submit some modules to BCR that somehow bring in the jars from Maven while also providing a BUILD file patch

What do those BUILD file patches do? What's the difference from using rules_jvm_external generated targets?

Reply:
sorry I wasn't clear -- I meant modules that require some sort of setup, like point 2 above described: 

> handles more complicated setups like Dagger and AutoValue where Java annotation processing tools are involved

plain Java library modules will not benefit from this, so are not in scope.

Reply:
I don't actually know how compelling bazel-common is for this repo specifically or even more generally nowadays, but a little extra info in case it helps you judge that:

- Part of what bazel-common does is help you release jars, Javadoc, pom files, etc. to Maven Central. If you don't need to do that, then that part won't help you.
- The dependency declarations of bazel-common were definitely more broadly useful [in the days before bzlmod](https://github.com/google/bazel-common/blob/9f8175287d46cdd21accb4b504a612b44c0f0564/workspace_defs.bzl).